fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/RequestedOrdering.java

```diff
@@ -305,7 +305,7 @@ public static RequestedOrdering ofPrimitiveParts(@Nonnull final List<RequestedOr
         Debugger.sanityCheck(() -> Verify.verify(
                 requestedOrderingParts.stream()
                         .allMatch(requestedOrderingPart -> requestedOrderingPart.getValue()
-                                .getResultType().isPrimitive())));
+                                .getResultType().isPrimitive() || requestedOrderingPart.getValue().getResultType().isUuid())));
```

Comment:
Sorry, I do not agree with this fix. The check makes sure that you never order by something that is not primitive. Isn't a UUID a structure? What happens if I want to order by a UUID column? That should be broken up just like other records should be broken up.